### ISSUE TITLE
v2.0.0 How to extend role declared in JSON?

### ISSUE THREAD

**Issue by eharrow** [onury/accesscontrol#22]
I can't seem to extend a role when declared in json.  The following works but when I take the tostring and then use as `access control.setGrants(thejson)` it complains about `$extend` being a reserved word.  I see that is a new addition in the changelog.  How do I achieve the same result using `setGrants()`?

```
const accessControl = new AccessControl();
accessControl.grant('viewer').readAny('users');
accessControl.grant('admin').extend('viewer').createAny('users').deleteAny('users').updateAny('users');
let assert = require('assert');
assert('admin can read any user=' + accessControl.can('admin').readAny('users').granted);
console.log(`accessControl.getGrants()=${JSON.stringify(accessControl.getGrants())}`);

$> accessControl.getGrants()={"viewer":{"users":{"read:any":["*"]}},"admin":{"$extend":["viewer"],"users":{"create:any":["*"],"delete:any":["*"],"update:any":["*"]}}}
```

**Comment by onury** [comment]
This looks like a bug. Will confirm and fix shortly.

**Comment by onury** [comment]
Fixed in v2.2.0.
